server/src/main/java/io/crate/analyze/where/DocKeys.java

```diff
@@ -102,7 +102,7 @@ public String getId(TransactionContext txnCtx, Functions functions, Row params,
             }
             return Lists2.map(
                 partitionIdx,
-                pIdx -> DataTypes.STRING.value(SymbolEvaluator.evaluate(txnCtx, functions, key.get(pIdx), params, subQueryResults)));
+                pIdx -> DataTypes.STRING.sanitizeValue(SymbolEvaluator.evaluate(txnCtx, functions, key.get(pIdx), params, subQueryResults)));
```

Comment:
Similar here. Partition by columns could have a different type, I'm not sure if `sanitizeValue` is enough here